netket/optimizer/jax/stochastic_reconfiguration.py

```diff
@@ -226,6 +245,41 @@ def compute_update(self, oks, grad, out=None):
 
         return out
 
+    def compute_update_onthefly(self, samples, grad, out=None):
+        r"""
+        Solves the SR flow equation for the parameter update ẋ.
+
+        The SR update is computed by solving the linear equation
+           Sẋ = f
+        where S is the covariance matrix of the partial derivatives
+        O_i(v_j) = ∂/∂x_i log Ψ(v_j) and f is a generalized force (the loss
+        gradient).
+
+        Args:
+            samples: An array of samples
+            grad: A pytree of the forces f
+            out: A pytree of the parameter updates that will be ignored
+        """
+
+        if self._x0 is None:
+            self._x0 = jax.tree_map(jnp.zeros_like, grad)  # x0 = jnp.zeros_like(grad)
+
+        if self._use_iterative:
```

Comment:
do we check that this works only if we are using the iterative solver?

Reply:
We check it in the init function: https://github.com/netket/netket/blob/b28b2ab264ace962f2561d9336e2272215373556/netket/optimizer/jax/stochastic_reconfiguration.py#L144

Reply:
great thanks